### ISSUE TITLE
Incorrect behaviour of safe navigation cop

### ISSUE THREAD

**Issue by nasospsa** [bbatsov/rubocop#3605]
When you want to create or update a model the safe navigation is triggered.

I havent found any related issue so that's why I'm making the issue.

```
model = Model.find id
if model
    model.update(atrs)
else
    Model.create(newAttrs)
end
```

---
## Expected behavior

This shouldn't raise an error
## Actual behavior

`Use safe navigation (&.) instead of checking if an object exists before calling the method`
## Possible solution

I have tried fixing by making the `if model` to `if model.present?` but I am not entirely happy with it. What do you think?
## RuboCop version

```
$ rubocop -V
0.43.0 (using Parser 2.3.1.4, running on ruby 2.3.1 x86_64-darwin15)
```

**Comment by bbatsov** [comment]
Try 0.44.1. I think this was fixed there.

**Comment by kluzny** [comment]
This also fails in `next if string.nil? || string.empty?` it recommends that I write this as string&.empty? but because empty? is a negative conditional it is not correct

```
irb(main):004:0> puts true if [].nil? || [].empty?
true
=> nil
irb(main):005:0> puts true if []&.empty?
true
=> nil
irb(main):006:0> puts true if nil.nil? || nil.empty?
true
=> nil
irb(main):007:0> puts true if nil&.empty?
=> nil
```

When the second condition is an affirmation  like rails' .present? the cop and recommendation make sense.

I think this is the same problem, but I'm open to making a new issue

**Comment by kluzny** [comment]
Actually my issues is addressed in https://github.com/bbatsov/rubocop/issues/3515, feel free to ignore me -_-

**Comment by tejasbubane** [comment]
I can confirm that this is no longer an issue with `0.45.0`. So this issue can be closed.

**Comment by nasospsa** [comment]
True, I am closing it :)
